File: Problem30/haskell/digit_fifth_powers.hs
digit_fifth_powers :: Integer
digit_fifth_powers = sum([x | x <- [2..200000], sum(fifth_powers(to_digits(x))) == x])

fifth_powers :: [Integer] -> [Integer]
fifth_powers [] = []
fifth_powers (x:xs) = x ^ 5 : fifth_powers xs

to_digits :: Integer -> [Integer]
to_digits 0 = []
to_digits n = n `mod` 10 : to_digits (n `div` 10)

File: Problem30/erlang/digit_fifth_powers.erl
-module(digit_fifth_powers).
-compile(export_all).

digitFifthPowers() ->
  lists:sum([X || X <- lists:seq(2,200000), lists:sum(fifthPowerDigits(X)) == X]).

fifthPowerDigits(0) -> [];
fifthPowerDigits(X) ->
  [round(math:pow((X rem 10), 5)) | fifthPowerDigits(X div 10)].
